packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java

```diff
@@ -554,14 +556,59 @@ private ConnectionStatusCallback requestConnectionStatusCallback(final Callback
         };
     }
 
+    @ReactMethod
+    public void showAndroidSoftKeyboard(Integer delayMillis) {
+        Activity currentActivity = mReactContext.getCurrentActivity();
+        if (currentActivity != null) {
+            View currentFocusedView = currentActivity.getCurrentFocus();
+            if (currentFocusedView != null) {
+                // Cancel any previously scheduled Runnable
+                if (mKeyboardRunnable != null) {
+                    currentActivity.getWindow().getDecorView().removeCallbacks(mKeyboardRunnable);
+                }
+
+                final Runnable showTheKeyboardNow = new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            View activeFocusedView = currentActivity.getCurrentFocus();
+                            if (activeFocusedView != null && currentActivity.getWindow().getDecorView().isShown()) {
```

Comment:
I have a question regarding the condition `currentActivity.getWindow().getDecorView().isShown()`, do we need to ensure that we don't show the keyboard when the editor is closed?

Reply:
Since we allow delaying this code to be run I thought it'd be safe to check the activity is still shown. Currently, the max value we use is `500` which is enough time I guess for the editor to be closed.

I did some tests when opening the Inserter menu and then quickly tapping on the back button twice to close the editor and this code ran when it was closed as it was already scheduled.

I also saw that if we pass the value of `activeFocusedView` as `null` the `InputMethodManager` would not try to show the keyboard but it looks like it can be different depending on the Android version where it might expect a value that is not `null`.

I'd suggest keeping these checks but I would be happy to investigate further if we can remove it. What do you think?

Reply:
> I did some tests when opening the Inserter menu and then quickly tapping on the back button twice to close the editor and this code ran when it was closed as it was already scheduled.

Ah, good to know that this case can be reproduced.

> I'd suggest keeping these checks but I would be happy to investigate further if we can remove it. What do you think?

My comment was merely a question to understand the code, I didn't want to imply that we should remove the code, so makes sense to me keeping the code 👍 .